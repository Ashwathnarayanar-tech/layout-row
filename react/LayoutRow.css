.textBlockContainer{
    padding: 20px;
}
.textBlockContainer h2,.textBlockContainer p{
    margin: 0;
    padding:20px 20px 20px 20px;
    line-height: 1.5;
}
.textBlockHeading,.textBlockPara{
    color: white;
    font-family: sans-serif;
}

.imageLink{
    margin: 0;
    padding: 0;
}

@media (max-width: 780px){
    
    .textBlockContainer h2,.textBlockContainer p{
        margin: 0;
        padding:10px 10px 10px 10px;
        line-height: 1.3;
    } 
}

@media (max-width: 480px){
    .textBlockContainer{
        display:block !important;
        padding: 0px ;
    }   
}