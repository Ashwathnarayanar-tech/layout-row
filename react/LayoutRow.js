import React from "react";
import styles from "./LayoutRow.css";
import { useCssHandles } from 'vtex.css-handles';
const CSS_HANDLES = [ 'LayoutRowContainer','textBlockContainer','textContainer','textBlockHeading','textBlockPara','textBlockImage','imageLink','bannerImage']
const LayoutRow = (props) => {
    const handles = useCssHandles(CSS_HANDLES)
const customStyle={
    textAlign: `${props.textBlockProperties.textAlign}`,
    color: `${props.textBlockProperties.textColor}`,
    backgroundColor:`${props.textBlockProperties.textBackground}`
    }    
    

    let blockText;
    if (props.textBlockProperties.textImage) {
        blockText = <div style={{...customStyle,display:"flex"}} className={handles.textBlockContainer} > 
                    <div style={{"flexFlow":"row"}} className={handles.textContainer} > 
                 <h2 className={handles.textBlockHeading}>{props.textBlock}</h2>
                <p className={handles.textBlockPara}>{props.textContent}</p>
                </div>
                <div>
                <img src={props.textBlockProperties.textImage} alt="Text Block Image"
                    style={{
                        height: `${props.textBlockProperties.textImageHeight}` ? `${props.textBlockProperties.textImageHeight}` : "200px",
                        // height: `${props.textBlockProperties.textImageWidth}` ? `${props.textBlockProperties.textImageWidth}` : "250px"
                    }}
                    className={handles.textBlockImage}/>
            </div>  
           </div> 
                           
        }
    else {
        blockText=  <div style={{...customStyle}} className={handles.textBlockContainer}> 
        <div className={handles.textContainer}> 
        <h2 className={handles.textBlockHeading}>{props.textBlock}</h2>
        <p className={handles.textBlockPara}>{props.textContent}</p>
        </div>            
    </div>
     }       
    
    return (
        <div className={handles.LayoutRowContainer}>      
          {!props.imageBlockProperties.positionTop && <div>{blockText}</div>}
            
            <a href={props.imageBlockLink} className={handles.imageLink}>           
                <img src={props.imageBlockSrc} alt="Banner Image"
                style={{
                    width: `${props.imageBlockProperties.fullWidth}` === 'true' ? "100%" : "",
                    height: `${props.imageBlockProperties.imageHeight}`
                    }}
                    className={handles.bannerImage}/>
            </a>            
            
            {props.imageBlockProperties.positionTop && <div>{blockText}</div>}
            
    </div>)
}

export default LayoutRow;