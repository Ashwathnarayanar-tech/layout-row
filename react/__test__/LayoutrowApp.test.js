import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import LayoutRow from '../LayoutRow'

configure({ adapter: new Adapter() })

describe('Element', () => {
  const element = shallow(
    <LayoutRow
      textContent="A 2020 Nissan Car"
      textBlockProperties={{
        textImage: 'https://echidna.vtexassets.com/arquivos/engine_135.png',
        textImageHeight: '200px',
        textImageWidth: '450px',
        textAlign: 'center',
        textColor: 'white',
        textBackground: 'black',
      }}
      imageBlockSrc="https://echidna.vtexassets.com/arquivos/banner_135.png"
      imageBlockLink="/"
      imageBlockProperties={{
        fullWidth: true,
        positionTop: true,
        imageHeight: '300px',
      }}
    />
  )
  test('Image Element render Test', () => {
    const ImgElementExists = element.find('div a img').exists()
    expect(ImgElementExists).toBe(true)
  })
  test('Text Element render Test', () => {
    const TxtElementExists = element.find('div div div h2').exists()
    expect(TxtElementExists).toBe(true)
  })
  test('Text Heading content check', () => {
    const textHeading = element.find('div div div p')
    expect(textHeading.text()).toBe('A 2020 Nissan Car')
  })

  test('should render in "debug" mode', () => {
    const component = shallow(
      <LayoutRow
        debug
        textContent="A 2020 Nissan Car"
        textBlockProperties={{
          textImage: 'https://echidna.vtexassets.com/arquivos/engine_135.png',
          textImageHeight: '200px',
          textImageWidth: '450px',
          textAlign: 'center',
          textColor: 'white',
          textBackground: 'black',
        }}
        imageBlockSrc="https://echidna.vtexassets.com/arquivos/banner_135.png"
        imageBlockLink="/"
        imageBlockProperties={{
          fullWidth: true,
          positionTop: true,
          imageHeight: '300px',
        }}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
