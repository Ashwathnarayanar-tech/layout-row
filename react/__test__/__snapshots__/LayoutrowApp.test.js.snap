// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element should render in "debug" mode 1`] = `ShallowWrapper {}`;
